### ISSUE TITLE
Unable to deploy PowerNSX with script using PowerCLI 6.5.1

### ISSUE THREAD

**Issue by m80arm** [vmware/powernsx#206]
When attempting to install PowerNSX via the following script:

$Branch="v2";$url="https://raw.githubusercontent.com/vmware/powernsx/$Branch/PowerNSXInstaller.ps1"; try { $wc = new-object Net.WebClient;$scr = try { $wc.DownloadString($url)} catch { if ( $_.exception.innerexception -match "(407)") { $wc.proxy.credentials = Get-Credential -Message "Proxy Authentication Required"; $wc.DownloadString($url) } else { throw $_ }}; $scr | iex } catch { throw $_ }

It throws up an error saying that PowerCLI is not installed even though PowerCLI 6.5.1 is installed

![powercli](https://cloud.githubusercontent.com/assets/26905652/25337102/5b02ecb6-28f2-11e7-8b95-d7470449f43f.jpg)

**Comment by pandom** [comment]
Further comments:

This was installed by Powershell Gallery - new to PowerCLI in 6.5.1.

This is the preffered method of deploying PowerNSX moving forward.

**Comment by dcoghlan** [comment]
The one liner installer currently installs using the V2 branch which is where the issue lies. The master branch currently does not suffer the same issue.

If you want to install from the master branch you can use the following one liner until we resolve the issue.
```
$Branch="master";$url="https://raw.githubusercontent.com/vmware/powernsx/$Branch/PowerNSXInstaller.ps1"; try { $wc = new-object Net.WebClient;$scr = try { $wc.DownloadString($url)} catch { if ( $_.exception.innerexception -match "(407)") { $wc.proxy.credentials = Get-Credential -Message "Proxy Authentication Required"; $wc.DownloadString($url) } else { throw $_ }}; $scr | iex } catch { throw $_ }
```

**Comment by nmbradford** [comment]
thanks @m80arm, and @dcoghlan for workaround - interesting the installers are detecting differently, its been a while since I looked at them - did you pinpoint where they differ?  I can backport any installer changes to v2 so it works...

**Comment by dcoghlan** [comment]
@nmbradford - V2 is looking for the registry key to uninstall the PowerCLI application to determine whether it is installed or not.

```
function check-powercli {


    #Validate at least PowerCLI 5.5 via uninstall reg key
    Write-Progress -Activity "Installing PowerNSX" -Status "check-powercli" -CurrentOperation "Checking for compatible PowerCLI version"
    if ((gwmi win32_operatingsystem).osarchitecture -eq "64-bit") { 
        $PowerCli = get-childitem "HKLM:Software\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall" | % { $_ | get-itemproperty | ? { $_.displayName -match 'PowerCLI' }}

    }else {
        $PowerCli = get-childitem "HKLM:Software\Microsoft\Windows\CurrentVersion\Uninstall" | % { $_ | get-itemproperty | ? { $_.displayName -match 'PowerCLI' }}
    }
```
The guidance for PowerCLI 6.5.1 is to uninstall the previous version, which removes this registry key. The install of the VMware.Powercli from Powershell Gallery is installed as a proper module now, so there is no registry key to look for.

Once PowerCLI is installed from Powershell Gallery though, the module is auto loaded so you can check the version by doing the following which will give you the version number
```
PS C:\Windows\system32> get-module VMware.Powercli -ListAvailable


    Directory: C:\Program Files\WindowsPowerShell\Modules


ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Manifest   6.5.1.5... VMware.PowerCLI
```

**Comment by nmbradford** [comment]
ah yes - reason for the messy way its happening currently is due to previous support for powercli 5.5 which was snapin based, not module.  Will resolve in 2.2 timeframe with changes to Powershell gallery based distribution.  Will be fixed in #239

**Comment by nmbradford** [comment]
This is resolved as of v3.0 release.
